controllers/validator_test.go

```diff
@@ -514,9 +515,16 @@ func TestDPAReconciler_ValidateDataProtectionCR(t *testing.T) {
 					},
 				},
 			},
-			objects: []client.Object{},
-			wantErr: true,
-			want:    false,
+			objects: []client.Object{
+				&corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cloud-credentials-gcp",
+						Namespace: "test-ns",
+					},
+				},
+			},
```

Comment:
Looks like we want to allow gcp without secrets for when they are using
`Option 2: Using Workload Identity` which is applicable when they are on GKE cluster.